gnupg/gnupg.py

```diff
@@ -558,6 +590,37 @@ def _parse_keys(self, result):
             if keyword in valid_keywords:
                 getattr(result, keyword)(L)
 
+    def extend_key(self, keyid, validity='1y', passphrase=None, extend_subkey=True):
+        """Extends a GnuPG key by passing in new validity period (from now) through
+            subprocess's stdin
+
+        >>> import gnupg
+        >>> gpg = gnupg.GPG(homedir="doctests")
+        >>> key_input = gpg.gen_key_input()
+        >>> key = gpg.gen_key(key_input)
+        >>> gpg.extend_key(key.fingerprint, '2w', 'good passphrase')
+
+        :param str keyid: key shortID, longID, email_address or fingerprint
+        :param str validity: 0 or number of days (d), or weeks (*w) , or months (*m) or years (*y)
+                               to extend the key, from its creation date.
```

Comment:
`validity` is a concept by itself in gnupg related to the trust of the key (https://www.gnupg.org/gph/en/manual/x334.html).

Maybe we can come up with a better name for this param. time? expiration_time?